### ISSUE TITLE
edit but don't overwrite configure_xxx.go

### ISSUE THREAD

**Issue by casualjim** [go-swagger/go-swagger#397]
currently keeping configure_xxx.go in sync is a manual exercise which is often times confusing.
It would be better if missing things could be added. 
Deleted operations and such should also be reflected in the configure_xxx.go file

Also track a list of generated files or use a .swagger.go extension so that we can delete all ephemeral files without worrying and we don't leave stale code lying around.

Updates should be ignored though, hopefully they'll fail compilation.

**Comment by ontucker** [comment]
I'm still getting used to how things in go-swagger are organized, but my desire is to have the API implementation live in an entirely different module so that I don't have to edit any of the files that live amongst the generated code.  I am currently modifying configure_xxx.go to just do a one-line call out to a server which exports a .ConfigureAPI method and then implements all the XYZHandlerFuncs itself and hooks them to 'api'.  However, it might be cleaner if the generated API defined an interface that contains all the handlers I need to provide and I could just pass something which implements that into operations.NewXXXAPI from main.

**Comment by AnsonT** [comment]
I started putting operation implementations under restapi/implementations. It would be great if configure_***.go file automatically binds to implementations.xxx methods when models are updated. The generator can add stub implementation if the implementation files do not exist, but leave existing ones in place.

**Comment by fredbi** [comment]
I had not seen this issue and just realized I've made the same on my own custom templates.

This could indeed be added as a featured CLI option (something like `--auto-wire-implementation`) but we should be wary of 2 things:
- the configurable implementation target package should be pre-existing, or the goimport won't work at generation time (this makes yet another `GenOpt` / CLI opt, like `--implementation="mypackage"`)
- the target handler is not arbitrarily and should be named after the operationID

In my own `configure_api.gotmpl`, this gives something like:
```golang
...
  // Registers server shutdown method
  api.ServerShutdown = func() {  }

  // Registers API operations
  {{- range .Operations}}
  api.{{if ne .Package $package}}{{pascalize .Package}}{{end}}{{pascalize .Name}}Handler = {{.Package}}.{{pascalize .Name}}HandlerFunc(implementation.{{pascalize .Name}})

  {{ end }}

  return setupGlobalMiddleware(api.Serve(setupMiddlewares))
...
```

This kind of feature is actually a direct application of using custom templates. Adding it as a feature should be rather straightforward. Feel free to submit a PR on this.

**Comment by gregmarr** [comment]
Yeah, that would work perfectly for my server, which uses the pattern:

	api.<name>Handler = operations.<name>HandlerFunc(routes.<name>)

I also customize `setupGlobalMiddleware` and I think `api.AccessCodeAuth`.

What might also be useful if rewriting is too complicated, is if `configure_<service>.go` already exists, we write out `configure_<service>.example` instead, so we can manually compare the two files.

**Comment by fredbi** [comment]
That would be interesting.

For my part, I worked around this generation issue with a script toolchain, which takes care of moving and merging the files, performing a diff3 between previous generation, current generation and current file... I made this because I have custom templates that go farther by creating a skeleton for my operations, resolving all params and producing the typical workflow I have with my database. So I can safely regenerate my skeletons one I have started to edit them manually. This works the same for the configure_api...

**Comment by e-nikolov** [comment]
@fredbi Do you generate stubs in the `implementation` package via the custom templates feature of go-swagger, or do you manually create it? I'm trying to do something similar, but I'm not sure how to tell go-swagger to put the generated files in a separate custom directory.

**Comment by casualjim** [comment]
You can use the template config file to that end: https://goswagger.io/use/template_layout.html#server-generation

**Comment by e-nikolov** [comment]
@casualjim I see. https://goswagger.io/generate/templates.html#use-custom-templates This part of the documentation confused me because it says that the directory part will become a part of the file name like so: server/test.gotmpl -> serverTest

**Comment by fredbi** [comment]
@e-nikolov I am using custom templates.

**Comment by casualjim** [comment]
@e-nikolov the custom-templates predate the config driven approach, we kept them so that we wouldn't break codebases who were already using them

**Comment by fredbi** [comment]
Resolved by #2510

**Comment by Joe1the2Creator3** [comment]
https://api.github.com/repos/Joe1the2Creator3/refactored-lamp/issues/3
